Branch-TestBed/Branch-SDK-Tests/BNCClassSerializationTests.m

```diff
@@ -0,0 +1,120 @@
+//
+//  BNCClassSerializationTests.m
+//  Branch-SDK-Tests
+//
+//  Created by Ernest Cho on 3/28/24.
+//  Copyright © 2024 Branch, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BranchEvent.h"
+#import "BranchOpenRequest.h"
+#import "BranchInstallRequest.h"
+
+@interface BranchEvent()
+// private BranchEvent methods used to build a BranchEventRequest
+- (NSDictionary *)buildEventDictionary;
+@end
+
+@interface BranchOpenRequest()
+- (NSString *)getActionName;
+@end
+
+@interface BNCClassSerializationTests : XCTestCase
+
+@end
+
+// Test serialization of replayable requests
+@implementation BNCClassSerializationTests
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+// BranchEventRequest is creation is tightly coupled with the BranchEvent class
+// In order to test building it, we need to expose some private methods. :(
+- (BranchEventRequest *)buildBranchEventRequest {
+    BranchEvent *event = [BranchEvent standardEvent:BranchStandardEventPurchase];
+    NSURL *url = [NSURL URLWithString:@"https://api3.branch.io/v2/event/standard"];
+    NSDictionary *eventDictionary = [event buildEventDictionary];
+    
+    BranchEventRequest *request = [[BranchEventRequest alloc] initWithServerURL:url eventDictionary:eventDictionary completion:nil];
+    return request;
+}
+
+- (void)testBranchEventRequestArchive {
+    BranchEventRequest *request = [self buildBranchEventRequest];
+
+    // archive the event
+    NSError *error = nil;
+    NSData *data = [NSKeyedArchiver archivedDataWithRootObject:request requiringSecureCoding:YES error:&error];
```

Comment:
`archivedDataWithRootObject` is deprecated https://developer.apple.com/documentation/foundation/nskeyedarchiver/1413189-archiveddatawithrootobject

Is there a replacement or is this fine for this purpose?

Reply:
Objective-C's odd function naming where arguments break up the function name gets people all the time.

https://developer.apple.com/documentation/foundation/nskeyedarchiver/2962880-archiveddatawithrootobject